build/version.cake

```diff
@@ -0,0 +1,71 @@
+public class BuildVersion
+{
+    public string Version { get; private set; }
+    public string SemVersion { get; private set; }
+    public string Milestone { get; private set; }
+    public string CakeVersion { get; private set; }
+
+    public static BuildVersion CalculatingSemanticVersion(
+        ICakeContext context,
+        bool isLocalBuild,
+        bool isPublishBuild,
+        bool isReleaseBuild
+        )
+    {
+        if (context == null)
+        {
+            throw new ArgumentNullException("context");
+        }
+
+        string version = null;
+        string semVersion = null;
+        string milestone = null;
+
+        if (context.IsRunningOnWindows())
+        {
+            context.Information("Calculating Semantic Version");
```

Comment:
@devlead minor point, but when writing messages to output, if you are "away" to do something, I prefer to see this as "Calculating Semantic Version..." The ... indicating that it is away to start. Thoughts?